DependencyInjection/DoctrineExtension.php

```diff
@@ -849,6 +851,14 @@ private function loadMessengerServices(ContainerBuilder $container) : void
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
         $loader->load('messenger.xml');
 
+        if (class_exists(DoctrineCloseConnectionMiddleware::class)) {
```

Comment:
Since the services are private and abstract, they'd be removed from the container if no-one uses them, right? If so, it would be better to just move both services into `messenger.xml` instead of conditionally loading extra service configs.

Reply:
Is it ok if some updates the bundle, he has to update messenger component (for example 4.2 to 4.3)?

Reply:
@alcaeus: updated but it depends on symfony PR